scripts/ci/lib.sh

```diff
@@ -1368,6 +1373,22 @@ EOT
     hub-comment -type build -template-file "$tmpfile"
 }
 
+junit2slack() {
+    info "Converting junit failures to slack attachments"
+
+    if [[ -z "${ARTIFACT_DIR}" ]]; then
+        info "Warning: junit2slack() requires an ARTIFACT_DIR"
+        return
+    fi
+
+    local junit_file_names
+    junit_file_names=$(find "${ARTIFACT_DIR}" -type f -name 'junit-*.xml' | xargs)
```

Comment:
this might need some extended bash globbing to get to junit xml in sub dirs. e.g. with this one:
https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/branch-ci-stackrox-stackrox-master-merge-local-roxctl-tests/1582096454128766976
there is a report.xml in 
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/branch-ci-stackrox-stackrox-master-merge-local-roxctl-tests/1582096454128766976/artifacts/merge-local-roxctl-tests/stackrox-initial/artifacts/junit-reports/
I don't think that `find` will find it.

Reply:
This actually brings up an interesting point (along with your suggestion on how to test these changes); at least from what it seems in the failed job [#1582162936795762688](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/branch-ci-stackrox-stackrox-master-merge-local-roxctl-tests/1582162936795762688), spyglass only looks at `junit_operator.xml`. In fact, when I tried to parse `merge-local-roxctl-tests/stackrox-initial/artifacts/junit-reports/report.xml`, it threw an error which leads me to believe we're not properly formatting some of these reports. That said, I haven't dug into that error yet. It could actually be the case that spyglass isn't surfacing useful error messages.

So then, should we only look at `report.xml` if present for now? Or spend the time to debug these (seemingly) malformed junit reports and try to parse all XML files? Interested in your take on this.

Reply:
> So then, should we only look at report.xml if present for now?

Different jobs/tests name the JUnit differently. Sometimes single files, sometimes directories of XML (or maybe a directory with a single file). There is a wrapper script `store_test_results` to give you an idea of what you are dealing with.

[#1582162936795762688](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/branch-ci-stackrox-stackrox-master-merge-local-roxctl-tests/1582162936795762688) is a special but interesting case. The prow/spyglass Go code must be hitting the same error that you are and that is why it is not displayed in the UI. For errors we should do the same, silently drop them from the slack message. But it would be good to look at fixing the XML when it is dropped and/or fixing the parser in slack2junit and/or prow.

There are quite a few cases where you'll see prow/spyglass parse the Junit we (stackrox tests) create OK: 
https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/branch-ci-stackrox-stackrox-master-merge-gke-qa-e2e-tests/1582357585149825024
https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/branch-ci-stackrox-stackrox-master-merge-openshift-penultimate-qa-e2e-tests/1580969317434920960

Reply:
Awesome, appreciate the feedback. I'll take a look at what's causing the malformed XML after this PR is merged.

---

Going back to the bash glob:
> this might need some extended bash globbing to get to junit xml in sub dirs

I thought `find` would do this automatically. From the POSIX manual:
> The find utility shall recursively descend the directory hierarchy from each file specified by path

Reply:
not with GNU find 4.9.0:
```
$ mkdir -p ttt/junit-crud
$ touch ttt/junit-crud/report.xml
$ find ttt -type f -name 'junit-*.xml'
$
```

Reply:
`**` will work though and does not seem to require `shopt extglob` (at least for my shell which is closer to the CI OS then say MacOS).
```
$ x=ttt/**/*.xml
$ echo $x
ttt/junit-crud/report-2.xml ttt/junit-crud/report.xml
$
```

Reply:
Circling back to this since I ended up going with `find`; it looks like your example is looking for `junit-*.xml` but the name of the file is `report.xml`.